service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CarePlanMapperTest.java

```diff
@@ -0,0 +1,170 @@
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01StructuredBody;
+import uk.nhs.connect.iucds.cda.ucr.ST;
+import uk.nhs.connect.iucds.cda.ucr.StrucDocContent;
+import uk.nhs.connect.iucds.cda.ucr.StrucDocText;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CarePlanMapperTest {
+    private static final String LANG = "EN";
+    private static final String SNOMED = "2.16.840.1.113883.2.1.3.2.4.15";
+    private static final String INFORMATION_ADVICE_GIVEN = "1052951000000105";
+    private static final String DESCRIPTION = "description";
+    private static final String DIV_AS_STRING = String.format("<div xmlns=\"http://www.w3.org/1999/xhtml\">%s</div>", DESCRIPTION);
+    private static final String TITLE = "title";
+
+    @Mock
+    private POCDMT000002UK01ClinicalDocument1 clinicalDocument;
+
+    @Mock
+    private Encounter encounter;
+
+    @Mock
+    private Period period;
+
+    @Mock
+    private POCDMT000002UK01Section section;
+
+    @Mock
+    private POCDMT000002UK01Component5 component5;
+
+    @Mock
+    private ST title;
+
+    @Mock
+    private CE code;
+
+    @Mock
+    private CS cs;
+
+    @Mock
+    private StrucDocText strucDocText;
+
+    @InjectMocks
+    private CarePlanMapper carePlanMapper;
+
+    @Before
+    public void setup() {
+        POCDMT000002UK01Component2 component2 = mock(POCDMT000002UK01Component2.class);
+        POCDMT000002UK01Component3 component3 = mock(POCDMT000002UK01Component3.class);
+        POCDMT000002UK01StructuredBody structuredBody = mock(POCDMT000002UK01StructuredBody.class);
+
+        code = CE.Factory.newInstance();
+        code.setCodeSystem(SNOMED);
+        code.setCode(INFORMATION_ADVICE_GIVEN);
+
+        when(clinicalDocument.getComponent()).thenReturn(component2);
+        when(component2.getStructuredBody()).thenReturn(structuredBody);
+        when(component3.getSection()).thenReturn(section);
+        when(component5.getSection()).thenReturn(section);
+        when(structuredBody.getComponentArray()).thenReturn(new POCDMT000002UK01Component3[] { component3 });
+
+        encounter.setPeriod(period);
+        title.setLanguage("EN");
+        cs.setCode("EN");
+        mockLanguageXmlStructure(cs);
+
+        Patient subject = mock(Patient.class);
+        Reference subjectReference = new Reference(subject);
+        when(encounter.getSubject()).thenReturn(subjectReference);
+        when(encounter.getSubjectTarget()).thenReturn(subject);
+        when(encounter.getPeriod()).thenReturn(period);
+    }
+
+    @Test
+    public void shouldMapITKReportToCarePlan() {
+        mockSection();
+
+        List<CarePlan> carePlans = carePlanMapper.mapCarePlan(clinicalDocument, encounter);
+        assertThat(carePlans).isNotEmpty();
+
+        CarePlan carePlan = carePlans.get(0);
+        assertThat(carePlan.getIdElement().getValue()).startsWith("urn:uuid:");
```

Comment:
can we extract this as a constant please

Reply:
Updated